downloadCurrentVersion.js

```diff
@@ -3,6 +3,8 @@
 // This is used to download the correct binary version
 // as part of the prepublish step.
 
+const SOLC_REPO = process.env.SOLC_REPO || 'https://raw.githubusercontent.com/ethereum/solc-bin/gh-pages/bin';
```

Comment:
My main concern with this override is that compiler hashes are not enforced. This means it would be possible that maliciously someone keeps downloading tainted compiler versions when using truffle, for example.

Is the use of private repositories of solc-bin a common thing? Is it still needed after the change to raw.githubusercontent?

A better solution would be to enforce hashes of downloaded compilers, but then we'd need to bake in the hash of `list.json` tied to a given version and nightlies could not be downloaded (well it would be flagged as untrusted).

Reply:
>My main concern with this override is that compiler hashes are not enforced.
My thinking was to enforce compiler hashes directly within your control to stop them being tainted. For example, if I was to setup an internal solc internal repo, all I would do is directly copy `solc-bin` and host it elsewhere. There's still potential when using GitHub that the maintainers of the repo update previous checksums and compiler versions.

>Is the use of private repositories of solc-bin a common thing? Is it still needed after the change to raw.githubusercontent?
I'd say it's common in the corporate world, but I've not seen it too much in ETH dev? For example, how tools like Artifactory can mirror RPM/npm repos to be used internally or simply just as a cache to be more redundant.

Overall, I think the main benefit of being able to use overrides is that teams could use custom solc versions which they host internally then hook in to tools like Truffle.

Reply:
> Overall, I think the main benefit of being able to use overrides is that teams could use custom solc versions which they host internally then hook in to tools like Truffle.

I'm not sure we should encourage the use of unofficial compiler versions. Do you have a reason in mind for an unofficial compiler build? The project has always tried to be accommodating to external input, which has led to avoiding forks of the compiler.